### ISSUE TITLE
Impl error on type in trait.

### ISSUE THREAD

**Issue by cmester0** [hacspec/hax#370]
The following:
```rust
pub trait Foo {
  type bar;
}
```
results in the error
```bash
Fatal error: exception Failure("impl_expr_atom: Todo implsource::param \n\nBinder(\n    <Self as Group>,\n    [],\n)")
```

**Comment by cmester0** [comment]
This seems to be an issue with `predicates` being empty in the `ImplSource::Param` branch of `frontend/exporter/src/traits.rs`.

**Comment by W95Psp** [comment]
Hi, yes I got that error at some point as well :(
This issue is caused by some constraint being of the shape `<Self as Foo>::bar : SomeTrait`: `Self` here acts as a local marker, so that we can refer to the associated type `bar`.
I guess your fix in PR #371 was to get rid of the error?
We need a `Self` keyword in sourceimpl I think. I can push a PR!
